yt_dlp/extractor/safari.py

```diff
@@ -23,32 +20,23 @@ class SafariBaseIE(InfoExtractor):
     LOGGED_IN = False
 
     def _perform_login(self, username, password):
-        _, urlh = self._download_webpage_handle(
-            'https://learning.oreilly.com/accounts/login-check/', None,
-            'Downloading login page')
-
-        def is_logged(urlh):
-            return 'learning.oreilly.com/home/' in urlh.geturl()
+        def is_logged():
+            result = self._download_json_handle('https://api.oreilly.com/api/v2/me/', None,
+                                                'Checking if logged in',
+                                                fatal=False, headers={'Accept': 'application/json'})
+            return result is not False
```

Comment:
since this is only used once, does it even need to be a function?

Reply:
what does that have to do with this being a function or not?

Reply:
It is called twice.

Reply:
ah, my bad